SDL/SdlParser.cpp

```diff
@@ -27,25 +29,34 @@ static const char* s_toknames[] = {
     "<Identifier>", "<Number>", "<String>", "<Typename>",
 };
 
-bool SDL::Parser::open(const char* filename)
+DS::Stream* SDL::Parser::stream() const
 {
-    char sanitycheck[12];
+    if (m_encStream)
+        return m_encStream;
+    else
+        return m_fileStream;
+}
 
-    m_file = fopen(filename, "r");
-    if (!m_file) {
-        ST::printf(stderr, "[SDL] Error opening file {} for reading\n", filename);
+bool SDL::Parser::open(const char* filename)
+{
+    m_fileStream = new DS::FileStream();
+    try {
+        m_fileStream->open(filename, "r");
+    } catch (DS::FileIOException& ex) {
+        ST::printf(stderr, "[SDL] Error opening file {} for reading: {}\n", filename, ex.what());
         return false;
```

Comment:
Leak (the previous version used `fopen(...) == NULL`, which meant no leaked handle was present)